crypto/bn/bn_mpi.c

```diff
@@ -45,7 +45,7 @@ BIGNUM *BN_mpi2bn(const unsigned char *d, int n, BIGNUM *ain)
     int neg = 0;
     BIGNUM *a = NULL;
 
-    if (n < 4) {
+    if (n < 4 || d[0] & 0x80) {
```

Comment:
Also this change may be an improvement but there is no test coverage for it,
which can be seen by the fact that the reported build failure is not affected by this change